### PR TITLE
Protect: add the ability to ignore threats for paid plan users

### PR DESCRIPTION
**Pull request by nateweller**
This PR adds the ability for paid users to ignore threats in Jetpack Protect.

#### Changes proposed in this Pull Request:

* Adds new static `Threats` class on the server side to help with managing and updating individual threats.
* Adds `/ignore-threat` endpoint to the Jetpack REST API.
* Adds new `<Dialog />` component, which syncs with the Redux store via `setDialog()` action.
* Adds new `<Notice />` component, which syncs with the Redux store via `setNotice()` action.
* Adds the ability to ignore a threat by sending a request to the wpcom API, via the `ignoreThreat()` action.
* Integrates all of the above into threat list items, to allow the ignore threat flow shown in the screenshots below.
* Includes a fix to display the scan in progress screen when the Scan API returns a `scanning` status.

**To Do:**
* Integrate the threat severity badge component once it is merged from #26533

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

1201069996155224-as-1203034583317302

p1HpG7-gSG-p2 

#### Does this pull request change what data or activity we track or use?

Nope!

#### Testing instructions:

* Boot up a fresh JN site with Jetpack Beta, running this branch of Protect
* Install a threat, using the Jetpack Debug tools or a vulnerable plugin such as [WooCommerce 3.0.0](https://downloads.wordpress.org/plugin/woocommerce.3.0.0.zip)
* Get started with Jetpack Protect free
* Validate no regression in the admin dashboard (free users should not be able to ignore threats).
* Upgrade to a Jetpack plan that includes scan.
* Validate the ignore threat flow as per the screenshots below.

#### Screenshots

Screen Shot 2022-10-05 at 9 54 22 AM

Screen Shot 2022-10-05 at 9 54 35 AM

Screen Shot 2022-10-05 at 10 00 53 AM

### CODE REVIEW

File: projects/plugins/protect/src/js/components/ignore-threat-modal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+				<div className={ styles.threat__severity }>
+					{ /* to do: implement severity badge component once available */ }
+					{ severity }
+				</div>
+			</div>
+
+			<Text mb={ 4 }>
+				{ createInterpolateElement(
+					__(
+						'By ignoring this threat you confirm that your have reviewed the detected code and assume the risks of keeping a potentially malicious file on your site. If you are unsure please request an estimate with <codeableLink>Codeable</codeableLink>.',
```
Comment (singerb) [inline projects/plugins/protect/src/js/components/ignore-threat-modal/index.jsx]: `your` -> `you`.

Beyond that, did this language come from the designs? Do we need to be quite so severe here? AFAIK we're not in Calypso, but maybe I missed that.

Comment (singerb) [reply projects/plugins/protect/src/js/components/ignore-threat-modal/index.jsx]: Also, as your screenshot demonstrates, the `potentially malicious` part is not always accurate. If we don't want to tailor this based on the threat, we should do something like `potentially malicious or vulnerable`.

Comment (nateweller) [reply projects/plugins/protect/src/js/components/ignore-threat-modal/index.jsx]: This text is taken directly from the design, which is copied from the Calypso confirmation:

Screen Shot 2022-10-06 at 9 26 58 AM

Screen Shot 2022-10-06 at 9 29 48 AM

I've updated the message via https://github.com/Automattic/jetpack/pull/26571/commits/441458dc03c091177f2a16ed2d11f7753ae0ba0a to fix the typo, and include the `or vulnerable` text to make it more widely applicable.

Comment (singerb) [reply projects/plugins/protect/src/js/components/ignore-threat-modal/index.jsx]: Yep, I actually found that later in the day by accident when I was searching for Codeable stuff. It's fine to match it here.

File: projects/plugins/protect/src/class-jetpack-protect.php [hunk 2]
```diff
@@ -260,6 +261,15 @@ public static function register_rest_endpoints() {
 				'callback' => __CLASS__ . '::api_clear_scan_cache',
 			)
 		);
+
+		register_rest_route(
+			'jetpack-protect/v1',
+			'ignore-threat',
+			array(
+				'methods'  => \WP_REST_SERVER::EDITABLE,
```
Comment (miguelxpn) [inline projects/plugins/protect/src/class-jetpack-protect.php]: We should probably add a capabilities check here to make sure the user is authenticated.

**Review by miguelxpn — CHANGES_REQUESTED** [review]

**Pull request merged**